lib/error-handler.js

```diff
@@ -62,13 +63,31 @@ function handleError (reply, error, cb) {
     return
   }
 
-  const result = func(error, reply.request, reply)
-  if (result !== undefined) {
-    if (result !== null && typeof result.then === 'function') {
-      wrapThenable(result, reply)
-    } else {
-      reply.send(result)
+  try {
+    const result = func(error, reply.request, reply)
+    if (result !== undefined) {
+      if (result !== null && typeof result.then === 'function') {
+        wrapThenable(result, reply)
+      } else {
+        reply.send(result)
+      }
+    }
+  } catch (err) {
+    reply[kReplyIsError] = true
+    try {
+      if (err.cause) {
+        throw new Error('Child error already has a cause set')
```

Comment:
Could you use a Fastify error with an error code?

Reply:
Done